.github/workflows/go-generate-check.yml

```diff
@@ -0,0 +1,50 @@
+name: Developer helper
+on: pull_request
+jobs:
+  go-generate-check:
+    name: Check if "go generate" has been run
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go 1.x
+        id: go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.16
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+      - name: Install deps
+        id: install-deps
+        run: |
+          SOLC_BIN=solc-linux-amd64-v0.6.12+commit.27d51765
+          curl -OL https://binaries.soliditylang.org/linux-amd64/$SOLC_BIN
+          sudo mv $SOLC_BIN /usr/bin/solc
+          sudo chmod +x /usr/bin/solc
+
+        shell: bash
+      - name: Install devtools
+        id: install-devtools
+        run: make devtools
+      - name: Run go:generate
+        id: go-generate
+        run: |
+          # ignore generating trezor package based on this comment:
+          # https://github.com/ethereum/go-ethereum/blob/master/accounts/usbwallet/trezor/trezor.go#L21-L43
+          go generate `go list ./... | grep -v trezor`
+      - name: Skip whitelisted generated go files
+        id: skip-files
+        run: |
+          # ignore `gen_genesis.go` file which has custom modifications for unmarshaling the various genesis formats
+          git checkout params/types/genesisT/gen_genesis.go
```

Comment:
@meowsbits here I suggest ignoring the `gen_genesis.go` file which has custom modification for core-geth support of different genesis flavours for each client.